feat: added on hover highlight for buttons

LindritRamusa opened the pull request.

LindritRamusa committed:
feat: added on hover highlight for buttons

LindritRamusa committed:
feat: added on hover highlight for buttons

Endrit-seek commented on frontend/src/assets/styles.css:
```diff
@@ -1,3 +1,24 @@
 @import 'tailwindcss/components';
 @import 'tailwindcss/utilities';
+
+.page-link,
```
Styles qe jan component based mos i qit ne global styles css. Next push i hek edhe i qet ne NavBar.vue

Endrit-seek commented on frontend/src/components/NavBar.vue:
```diff
@@ -5,19 +5,19 @@
 
         <div class="flex space-x-4">
           <router-link to="/login" class="text-white hover:text-gray-300">
-            <button class="text-white hover:text-gray-300">Login</button>
```
Login edhe Register buttons nuk duhet me pas kete efekt sikur menut e navbarit. Ne vend ketij efekti i ban me outline. Mundsisht vetem Login button me qene me kete outline effect, register mundet me qene veq plain pa effect on hover.
[Shembull ](https://pbs.twimg.com/media/Fl0cjwQX0AU6OWC?format=jpg&name=4096x4096)

LindritRamusa replied:
i editova ndryshimet edhe i bana push me commit-in tjeter, mundesh me ba merge me develop

Endrit-seek committed:
feat: created authentication store management

Created a type for User, added packages that we will need, added a gitignore missing for .env files, added a example env file, configured tsconfig to allow import env variables

Endrit-seek committed:
feat: designed and added login and register functionality

Added LottieFiles package and animation, configured router to not include default layout in login and register page, designed LoginView and RegisterView, added cors for API calls to include localhost, added lottie folder for future references.

elionbehluli committed:
feat: implemented user logout on backend

Endrit-seek committed:
feat: added pinia persisted state and register function

Endrit-seek committed:
feat: implemented register and cookie jwtToken

Endrit-seek committed:
feat: added dashboardview and route

Endrit-seek committed:
fix: duplicate code in types

Endrit-seek committed:
feat: added DeleteAPIResponse type

elionbehluli committed:
feat: finalized user forgot password on backend

Endrit-seek committed:
feat: implemented snackbar component and store

In addition to that, refactored all api calls to use snackbar for user notification once a api call is done. Also designed the login and register form with validation, to use snackbar, and made it visibly clear to the user when a api call is being sent to the server

Endrit-seek committed:
fix: small refactor in logout api

ValezaL committed:
feat: finalized seeders for brands and cars in backend

elionbehluli committed:
feat: finalized crud for cars in backend

elionbehluli committed:
fix: added AdminCheck middleware to update and destroy route

ValezaL committed:
feat: finalized forget and reset page in frontend

LindritRamusa committed:
fix: added active link highlighting in navbar and refactor button styles and hover effects

elionbehluli committed:
feat: finalized upload car images on backend

ValezaL committed:
feat: finalized about us page in frontend

Commit:
feat: merged with develop

Commit:
feat: merged
